SwiftUIHW/SwiftUIHW/ContentView.swift

```diff
@@ -7,21 +7,191 @@
-            Text("Hello, world!")
+        ZStack {
+            gradientBackround
+            VStack {
+                Spacer()
+                songImage
+                Spacer()
+                HStack {
+                    songNameText.padding([.leading], Constants.buttonStackPaddingValue).padding(.bottom, Constants.buttonImageHeight)
+                    Spacer()
+                }
+                songProgressSlider.animation(.easeIn, value: songDuration)
+                songTime
+                buttonStack.padding([.bottom], Constants.buttonStackPaddingValue)
+                volumeSlider
+                shareDownloadButtons
+            }
+        }
+    }
+    
+    // MARK: - Private properties
+    
+    @StateObject private var viewModel = PlayerViewModel()
+    
+    @State private var volumeValue: Float = Constants.zero
+    @State private var songDuration: Float = Constants.zero
+    @State private var isPlaying = false
+    @State private var isDownloadedShown = false
+    @State private var isShareAlertShown = false
+    
+    private var volumeSlider: some View {
+        Slider(value: Binding(get: {
+            viewModel.currentVolume
+        }, set: { newValue in
+            viewModel.currentVolume = newValue
+            viewModel.updateVolume(value: newValue)
+        }), in: Constants.zero...Constants.hundrid) {
+            
+        } minimumValueLabel: {
+            Image(systemName: Constants.lowSpeakerImageName).foregroundColor(.white)
+        } maximumValueLabel: {
+            Image(systemName: Constants.highSpeakerImageName).foregroundColor(.white)
+        }.padding([.leading, .trailing], Constants.sliderPaddingValue)
+    }
+    
+    private var songTime: some View {
+        HStack {
+            Spacer()
+            Text("\(viewModel.songTime())").padding([.bottom, .trailing], Constants.sliderPaddingValue)
+        }
+    }
+    
+    private var gradientBackround: some View {
+        LinearGradient(gradient: Gradient(colors: [Color(Constants.colorName), Color(Constants.gradientColorName)]), startPoint: .top, endPoint: .bottom).ignoresSafeArea()
+    }
+    
+    private var songProgressSlider: some View {
+        Slider(value: Binding(get: {
+            viewModel.currentTime
+        }, set: { newValue in
+            viewModel.currentTime = newValue
+            viewModel.setTime(value: newValue)
+        }), in: Constants.zero...viewModel.maxDuration).padding([.leading, .trailing], Constants.sliderPaddingValue)
+    }
+    
+    private var shareDownloadButtons: some View {
+        HStack {
+            Button {
+                isShareAlertShown = true
+            } label: {
+                Image(systemName: Constants.shareImageName).resizable()
+                    .frame(width: Constants.buttonImageWidth, height: Constants.buttonImageHeight)
+                    .foregroundColor(.white)
+            }.alert(isPresented: $isShareAlertShown) {
```

Comment:
```suggestion
            }
            .alert(isPresented: $isShareAlertShown) {
```
Все модифаеры с новой строки поправь везде